Sources/Regex/Parse/Token.swift

```diff
@@ -19,6 +19,9 @@ enum Token: Hashable {
   case setOperator(SetOperator)
   case character(Character, isEscaped: Bool)
   case unicodeScalar(UnicodeScalar)
+  case builtinCharClass(CharacterClass)
+  case anchor(Anchor)
+  case specialCharEscape(SpecialCharacterEscape)
```

Comment:
Have you thought about what the AST representation should be?

Reply:
Yeah I'm currently not too sure, it seems they could either be their own category or just be folded into character or unicode scalar. I don't really feel too strongly either way, I don't suppose you have any thoughts?